### PR TITLE
Setup Wizard feature flag + Local install global alert cleanup

### LINKED ISSUE #46513: Remove all unimportant to local setup global banners
At the moment, we show a lot of global nav banners when we run single docker container Sourcegraph instance. Some of them  are not even related to the docker container setup (like externalUIL setting) 

It seems like that for the single docker deploy type we should remove or replace 
- Connect to code host banner (because the new setup UI wizard will replace it) (instead of code host management page it should lead to the /setup route)
- Email smtp (because this setting isn't relevant for the Sourcegraph local setup)
- ExternalURL (because this also ins't relevant to local setup)

Screenshot 2023-01-16 at 10 35 27

Since our new setup UI isn't ready yet we should create a feature flag for the new setup UI and apply changes only if feature flag is on. 

When feature flag is off - so all banners should be like they are now
When feature flag is on - connect code host banner should lead to `/setup` and other should be hided

### PR DESCRIPTION
**Pull request by st0nebraker**
Closes #46513 papercut

- Adds a frontend feature flag for Setup-Experience project
- Cleans up global alerts for initial load on local install
    - `Connect code host` banner doesn't display (Eventually all admin setup experiences will rid of these, once the Setup Wizard is compatible)
    - Removes backend from setting global alert for `email.smtp` + `externalURL` if `deployType` is `docker-container`

**TL;DR:** Removes 3/4 alerts below for `Code host`, `email.smtp`, + `externalURL` for local install (`docker-container` deploy type)
image

## Test plan
- Boot up local env with an admin account missing code hosts
- Add `?feature-flag-key=setup-wizard&feature-flag-value=true` feature flag to url

### CODE REVIEW

File: cmd/frontend/graphqlbackend/site_alerts.go [hunk 1]
```diff
@@ -89,6 +90,9 @@ var disableSecurity, _ = strconv.ParseBool(env.Get("DISABLE_SECURITY", "false",
 
 func init() {
 	conf.ContributeWarning(func(c conftypes.SiteConfigQuerier) (problems conf.Problems) {
+		if deploy.IsDeployTypeSingleDockerContainer(deploy.Type()) {
+			return nil
+		}
```
Comment (st0nebraker) [inline cmd/frontend/graphqlbackend/site_alerts.go:95]: @gl-srgr please confirm my go changes look ok, albeit, they're small, I'm new to go 😬 . Basically, these 2 alerts should not be set if the `deployType` is `docker-container`

image

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.05 kb) | 0.00% (+0.05 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/becca/setup-wizard-feature-flag/compare-report.html;tab=live_object) for a full comparison between the commits b633b9c and 8c77dc5 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/featureFlags/featureFlags.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export type FeatureFlagName =
     | 'enable-streaming-git-blame'
     | 'plg-enable-add-codehost-widget'
     | 'enable-rbac'
+    | 'setup-wizard'
```
Comment (vovakulikov) [inline client/web/src/featureFlags/featureFlags.ts:19]: I think it would be better if we would used an experimental feature flag instead, they are loaded synchronously so it should be no flash while we're loading feature flag.

File: client/web/src/site/NeedsRepositoryConfigurationAlert.tsx [hunk 3]
```diff
@@ -17,15 +18,22 @@ const onClickCTA = (): void => {
...
-        <Link className="site-alert__link" to="/site-admin/external-services" onClick={onClickCTA}>
-            <span className="underline">Connect a code host</span>
-        </Link>
-        &nbsp;to connect repositories to Sourcegraph.
-    </DismissibleAlert>
-)
+> = ({ className }) => {
+    const [isSetupWizardEnabled] = useFeatureFlag('setup-wizard')
+
+    if (isSetupWizardEnabled) {
```
Comment (vovakulikov) [inline client/web/src/site/NeedsRepositoryConfigurationAlert.tsx:24]: I think it still be useful to have this banner but we need to change a link there 
instead of `/site-admin/external-services` redirection we should open the /setup page

File: client/web/src/featureFlags/featureFlags.ts (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/featureFlags/featureFlags.ts:19]: In this PR https://github.com/sourcegraph/sourcegraph/pull/46952 we also should use an experimental feature flag I believe

**Review by vovakulikov — COMMENTED** [review]
Left a few comments regarding feature flag and needed repository setup link

File: client/web/src/site/NeedsRepositoryConfigurationAlert.tsx (hunk 3 above)
Comment (st0nebraker) [reply client/web/src/site/NeedsRepositoryConfigurationAlert.tsx]: ✅  good call!

File: client/web/src/featureFlags/featureFlags.ts (hunk 2 above)
Comment (st0nebraker) [reply client/web/src/featureFlags/featureFlags.ts]: Gotcha, thanks. I agree, avoiding async here is better. Updated in [e72a276](https://github.com/sourcegraph/sourcegraph/pull/46816/commits/e72a2762d6c2e68484fcf2f47dd7be940c338cb1). Note the flag setting is set & route set in #46952

File: client/web/src/site/NeedsRepositoryConfigurationAlert.tsx [hunk 4]
```diff
@@ -17,15 +18,23 @@ const onClickCTA = (): void => {
...
-        className={classNames('d-flex align-items-center', className)}
-    >
-        <Link className="site-alert__link" to="/site-admin/external-services" onClick={onClickCTA}>
-            <span className="underline">Connect a code host</span>
-        </Link>
-        &nbsp;to connect repositories to Sourcegraph.
-    </DismissibleAlert>
-)
+> = ({ className }) => {
+    const isSetupWizardEnabled = !!getExperimentalFeatures().enableSetupWizard
```
Comment (vovakulikov) [inline client/web/src/site/NeedsRepositoryConfigurationAlert.tsx:22]: I think we can use just `useExperimentalFeatures` zustand hook store here

**Review by vovakulikov — APPROVED** [review]
FE part looks good to me, I left the BE review part to the @gl-srgr eyes

File: cmd/frontend/graphqlbackend/site_alerts.go (hunk 1 above)
Comment (gl-srgr) [reply cmd/frontend/graphqlbackend/site_alerts.go:95]: Looks good!
